@jfindlay I think the test needs to be modified to look for a list of grain values since list_absent now looks for a list of values.  The updates I made now always returns a list of values even if one grain value is passed.

```Traceback (most recent call last): File "/testing/tests/unit/states/grains_test.py", line 161, in test_list_absent ret1) AssertionError: {'comment': 'Value edam in grain cheese is set to be deleted', 'changes': {'dele [truncated]... != {'comment': 'Value edam in grain cheese is set to be deleted', 'changes': {'dele [truncated]... - {'changes': {'deleted': ['edam']}, ? - - + {'changes': {'deleted': 'edam'}, 'comment': 'Value edam in grain cheese is set to be deleted', 'name': 'cheese', 'result': None} ```

--- a/tests/unit/states/grains_test.py
+++ b/tests/unit/states/grains_test.py
@@ -142,7 +142,7 @@ class GrainsTestCase(TestCase):
         ret = {'changes': {}, 'name': self.name, 'result': True,
                'comment': 'Value edam is absent from grain cheese'}
 
-        ret1 = {'changes': {'deleted': self.value}, 'name': self.name,
+        ret1 = {'changes': {'deleted': [self.value]}, 'name': self.name,
                 'result': None,
                 'comment': 'Value edam in grain cheese is set to be deleted'}
 
@@ -157,7 +157,7 @@ class GrainsTestCase(TestCase):
 
         with patch.dict(grains.__opts__, {'test': True}):
             with patch.dict(grains.__grains__, {self.name: [self.value]}):
-                self.assertDictEqual(grains.list_absent(self.name, [self.value]),
+                self.assertDictEqual(grains.list_absent(self.name, self.value),
                                      ret1)
 
         self.assertDictEqual(grains.list_absent(self.name, self.value), ret2)

--- a/tests/unit/states/grains_test.py
+++ b/tests/unit/states/grains_test.py
@@ -157,7 +157,7 @@ class GrainsTestCase(TestCase):
 
         with patch.dict(grains.__opts__, {'test': True}):
             with patch.dict(grains.__grains__, {self.name: [self.value]}):
-                self.assertDictEqual(grains.list_absent(self.name, self.value),
+                self.assertDictEqual(grains.list_absent(self.name, [self.value]),
                                      ret1)
 
         self.assertDictEqual(grains.list_absent(self.name, self.value), ret2)

